plugins/modules/ndo_l3out_template.py

```diff
@@ -0,0 +1,967 @@
+    ospf = module.params.get("ospf")
+    bgp = module.params.get("bgp")
+
+    state = module.params.get("state")
+
+    l3out_template_object = MSOTemplate(mso, "l3out", l3out_template)
+    l3out_template_object.validate_template("l3out")
+
+    tenant_id = l3out_template_object.template_summary.get("tenantId")
+    tenant_name = l3out_template_object.template_summary.get("tenantName")
+
+    l3outs = l3out_template_object.template.get("l3outTemplate", {}).get("l3outs", [])
+
+    if state in ["query", "absent"] and l3outs == []:
+        mso.exit_json()
+    elif state == "query" and not (name or uuid):
+        mso.existing = l3outs
+    elif l3outs and (name or uuid):
+        object_description = "L3Out"
+
+        if uuid:
+            match = l3out_template_object.get_object_by_uuid(object_description, l3outs, uuid)
+        else:
+            match = l3out_template_object.get_object_by_key_value_pairs(object_description, l3outs, [KVPair("name", name)])
+
+        if match:
+            mso.existing = copy.deepcopy(match.details)
+            mso.previous = copy.deepcopy(match.details)
+
+    ops = []
+
+    if state == "present":
+        if uuid and not mso.existing:
+            mso.fail_json(msg="L3Out with the uuid: '{0}' not found".format(uuid))
+
+        templates_objects_path = "templates/objects"
+        route_map_params = {"type": "routeMap", "tenant-id": tenant_id}
+        route_map_path = generate_api_endpoint(templates_objects_path, **route_map_params)
+        route_map_objects = mso.query_objs(route_map_path)
+
+        vrf_ref = None
+        if vrf_dict:
+            vrf_object = l3out_template_object.get_vrf_object(vrf_dict, tenant_id, templates_objects_path)
+            if pim and vrf_object.details.get("l3MCast") is False:
+                mso.fail_json(
+                    msg="Invalid configuration in L3Out {0}, 'PIM' cannot be enabled while using the VRF '{1}' with L3 Multicast disabled".format(
+                        "UUID: {0}".format(uuid) if uuid else "Name: {0}".format(name), vrf_dict.get("name")
+                    )
+                )
+            vrf_ref = vrf_object.details.get("uuid")
+
+        existing_routing_protocols = mso.existing.get("routingProtocol", None)
+
+        ospf_state = ospf.get("state") if ospf else "ignore"
+        bgp_state = bgp.get("state") if bgp else "ignore"
+
+        res_routing_protocols = get_routing_protocol(None if existing_routing_protocols == "none" else existing_routing_protocols, ospf_state, bgp_state)
+
+        routing_protocols = res_routing_protocols if res_routing_protocols else "none"
+
+        if not mso.existing:
+            payload = dict(name=name)
+
+            payload["routingProtocol"] = routing_protocols
+
+            if vrf_ref is None:
+                mso.fail_json(msg="The O(vrf) is required during the creation.")
+
+            payload["vrfRef"] = vrf_ref
+
+            if description:
```

Comment:
This should likely be  
```
if description is not None:
    payload["description"] = description
```

Reply:
Empty string is not allowed during the creation.